bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManagerImpl.java

```diff
@@ -477,8 +485,16 @@ public void thingRemoved(final Thing thing, ThingTrackerEvent thingTrackerEvent)
             if (thingHandlerFactory != null) {
                 unregisterAndDisposeHandler(thingHandlerFactory, thing, thingHandler);
                 if (thingTrackerEvent == ThingTrackerEvent.THING_REMOVED) {
-                    safeCaller.create(thingHandlerFactory, ThingHandlerFactory.class).build()
-                            .removeThing(thing.getUID());
+                    caller.execSync(() -> {
+                        thingHandlerFactory.removeThing(thing.getUID());
+                    }, new ExecutionConstraints(SafeCaller.DEFAULT_TIMEOUT, () -> {
```

Comment:
Perhaps :wink:

IMHO as long as both variants exist it is simpler to use one constant, so everyone knows it is the identical one (it is ATM).
As soon as the timeouts differ, it must be changed -- and if the safe caller can be dropped because there is no need to create a proxy at all, we will change the naming at all.
But perhaps we will keep the safe caller for future use, also if it is not used anymore...?

So, ATM I assume it can confuse the user why there are two constants for the same "thing".